pytest_fixtures/component/provision_pxe.py

```diff
@@ -0,0 +1,226 @@
+import ipaddress
+
+import pytest
+from box import Box
+from broker.broker import Broker
+from fauxfactory import gen_string
+from packaging.version import Version
+
+from robottelo import constants
+from robottelo.api.utils import enable_rhrepo_and_fetchid
+from robottelo.api.utils import wait_for_tasks
+from robottelo.config import settings
+from robottelo.hosts import ContentHost
+
+
+@pytest.fixture(scope='module')
+def provisioning_capsule(module_target_sat, module_location):
+    """Assings the `module_location` to Satellite's internal capsule and returns it"""
+    capsule = (
+        module_target_sat.api.SmartProxy()
+        .search(query={'search': f'name={module_target_sat.hostname}'})[0]
+        .read()
+    )
+    capsule.location = [module_location]
+    capsule.update(['location'])
+    return capsule
+
+
+@pytest.fixture(scope='module')
+def provisioning_rhel_content(
+    request,
+    module_target_sat,
+    module_org_with_manifest,
+    module_lce_library,
+    module_default_org_view,
+):
+    """
+    This fixture sets up kickstart repositories for a specific RHEL version
+    that is specified in `request.param`.
+    """
+    rhel_ver = request.param
+    sat = module_target_sat
+    repo_names = [f'rhel{rhel_ver}']
+    if int(rhel_ver) > 7:
+        repo_names.append(f'rhel{rhel_ver}_aps')
+
+    rh_repos = []
+    tasks = []
+    for name in repo_names:
+        rh_repo_id = enable_rhrepo_and_fetchid(
+            basearch=constants.DEFAULT_ARCHITECTURE,
+            org_id=module_org_with_manifest.id,
+            product=constants.REPOS['kickstart'][name]['product'],
+            repo=constants.REPOS['kickstart'][name]['name'],
+            reposet=constants.REPOS['kickstart'][name]['reposet'],
+            releasever=constants.REPOS['kickstart'][name]['version'],
+        )
+        # Sync step because repo is not synced by default
+        rh_repo = module_target_sat.api.Repository(id=rh_repo_id).read()
+        task = rh_repo.sync(synchronous=False)
+        tasks.append(task)
+        rh_repos.append(rh_repo)
+    for task in tasks:
+        wait_for_tasks(
+            search_query=(f'id = {task["id"]}'),
+            poll_timeout=2500,
+        )
+        task_status = module_target_sat.api.ForemanTask(id=task['id']).poll()
```

Comment:
wait_for_tasks should return list of task_statuses so no need to call API again.

Reply:
`wait_for_tasks` returns a list of tasks. In my case, there is one task in the list. This task still has the result `pending`. That is why I have to do an extra API call. The implementation of `wait_for_tasks` would have to change.